### PR TITLE
DC-272: Integrate SiteImprove analytics for DC portal

### PR DESCRIPTION
**Pull request by adbergen**
#### 🔗 Jira ticket
https://codeforamerica.atlassian.net/browse/DC-272

#### ✍️ Description

Adds SiteImprove analytics to the DC portal so the DC team can collect page-view and event data through Siteimprove.

Two pieces:

1. **`@sebt/analytics` package** — new `siteimprove-bridge` that listens to the existing data-layer events (`PageViewed`, `EventTracked`) and pushes the corresponding `_sz` commands. Mirrors the shape of the existing Mixpanel and Amplitude bridges so all three vendors stay symmetric.
2. **Portal Web** — new `<SiteImproveAnalytics />` client component (Next.js `<Script>` with the request-scoped CSP nonce) wired into the root layout. Render is **DC-only**: gated on both `state === 'dc'` and the `NEXT_PUBLIC_SITEIMPROVE_ID` env var, so an env-var leak into another state won't enable it. CSP `connect-src`/`script-src` are allow-listed for `siteimproveanalytics.com` / `siteimproveanalytics.io` only when the env var is set. The CSP-construction block in `proxy.ts` was also refactored to build directives as filterable arrays — easier to scan now that we have three optional vendors (GA, Mixpanel, Amplitude, SiteImprove).

#### Mapping (per ticket acceptance criteria)

- **`action`** = SEBT event name (e.g. `cta_click`, `page_load`).
- **`category`** = derived from the data layer's `page.flow` and `page.step` (the per-route values registered in [analytics-routes.ts](src/SEBT.Portal.Web/src/lib/analytics-routes.ts) → `DataLayerProvider`). Combined as `flow:step` so analysts can segment by both (e.g. `auth:verify_otp`, `address_update:address_form`, `cards:info`). Falls back to whichever field is populated, then to `unknown` when an unmapped route renders.
- **`label`** = JSON-serialized `eventData` with known PII keys dropped (recursive). Deny-list covers email, phone, name, address, DOB, SSN, etc. — see `PII_KEYS` in `siteimprove-bridge.ts`. Defensive floor; per-event allow-list can replace this once DC confirms which fields they want.
- **Page views** use SiteImprove's `['trackdynamic', { url, ref, title }]` shape (the API for SPA route changes); custom events use `['event', category, action, label?]`.

#### 🔗 Links to related PRs
N/A

#### ✅ Completion tasks

- [x] Added relevant tests (11 unit tests on the bridge: action mapping, category from page.flow+page.step, flow-only fallback, unknown-route fallback, PII scrubbing including nested, label-omitted edge case, deferred-init, teardown)
- [x] Meets acceptance criteria in ticket
- [ ] Configuration changes:
  - [x] If new environment variables are added, update in Tofu — \`NEXT_PUBLIC_SITEIMPROVE_ID\` needs to be added to the DC Tofu config (DC only)
  - [ ] If new environment secrets are added, update in Tofu and set in AWS Secret Manager
  - [ ] If you're adding an appsetting, add it to the requisite example file, and update it in the AWS AppConfig
  - [ ] If appsetttings are changed, update in AWS AppConfig

### CODE REVIEW

**Comment by adbergen** [comment]
# QA Summary — DC-272 SiteImprove integration (DC)

## What Changed

The DC portal now loads the SiteImprove analytics tracker. Page views and tracked events (e.g. CTA clicks) flow into Siteimprove for the DC team to analyze. No user-visible UI change — QA verification is about confirming the tracker loads on DC, events reach Siteimprove, and nothing changes for CO. Strictly DC-only: the Colorado portal must show no SiteImprove activity at all.

## Affected Areas

- All DC portal pages — page views and click events
- Browser console / Network panel (presence of SiteImprove script and tracker requests)
- CO portal — regression: should remain untouched

## Test Cases

### Happy Path

1. **SiteImprove script loads on DC.** Open the DC portal in a browser. Open DevTools → Network → filter for "siteimprove". After the page settles, you should see a request to `siteimproveanalytics.com/js/siteanalyze_<id>.js` that returns 200. Reload the page and confirm the script loads on every page.
2. **Page views are tracked.** Sign in to the DC portal and navigate through several pages (dashboard → profile → cards → back to dashboard). For each navigation, DevTools → Network → filter for "siteimproveanalytics.io" should show a tracker request with the new URL. (If you have access to the SiteImprove dashboard, the same navigation should appear there within a few minutes.)
3. **CTA-click events are tracked.** From the dashboard, click a tracked CTA (e.g. "Update mailing address" or "Request a replacement card"). Network panel should show another tracker request to `siteimproveanalytics.io` shortly after the click.
4. **No console errors.** Throughout all of the above, the browser console should be free of any "Refused to connect" / CSP errors related to `siteimproveanalytics.com` or `siteimproveanalytics.io`.

### Edge Cases

1. **Spanish language.** Switch the language toggle to Español and navigate a page. The tracker should still fire — language toggle does not stop analytics.
2. **Slow network.** Throttle to "Slow 3G" and reload a page. The SiteImprove script load may take longer but should not be blocked or fail; navigation events should still queue and eventually deliver.
3. **Signed-out pages.** Open the DC portal sign-in page (without signing in). The SiteImprove tracker should still load — analytics covers public pages too.
4. **Direct URL navigation.** Paste a deep-link URL (e.g. `/cards/info`) into the address bar. The page-view tracker should fire after the page loads.

### Regression Checks

1. **CO portal sees nothing.** Open the CO portal. DevTools → Network panel filtered for "siteimprove" should show **zero requests**. The browser console should have no SiteImprove-related entries. This is the most important regression — DC-272 is intentionally DC-only.
2. **Other analytics still work.** On DC, confirm that Google Analytics 4, Mixpanel, and Amplitude (if configured for the environment) continue to fire as they did before this change. Checked via Network panel filters for `google-analytics.com`, `mixpanel.com`, and `amplitude.com` respectively.
3. **Sign-in still works.** Sign in via the identity provider on both DC and CO portals — auth flow should be unchanged. No new CSP errors during the redirect or callback.
4. **Socure ID-proofing still works.** On DC, trigger the document-verification flow if available. Socure's embedded SDK and its Datadog telemetry should still load without CSP errors.

## Environment Notes

- The DC portal must have `NEXT_PUBLIC_SITEIMPROVE_ID` set to the SiteImprove site ID for the tracker to load. Coordinate with the team running Tofu to confirm this is configured on the environment under test.
- Do **not** set `NEXT_PUBLIC_SITEIMPROVE_ID` on the CO portal — even if accidentally set, the code blocks rendering on CO (state gate), but the env var should be omitted in CO config for clarity.
- Verifying the data path end-to-end (events landing in Siteimprove's dashboard) requires access to the DC SiteImprove account.
- Frontend-only change — no API, database, or backend updates.

**Comment by adbergen** [comment]
Closed the two gaps Morgan's refined acceptance criteria called out (commit b9789530):

**1. `category` is now path-derived.** Was hardcoded to `'sebt-portal'`; now `deriveCategory()` returns the first non-empty path segment (`dashboard`, `cards`, `profile`, `login`, …) with `root` as the fallback for `/`. Pragmatic read of "page/flow context" — `page.flow`/`page.step` don't actually exist in our data layer, so pathname is the closest equivalent we can wire today. Per-page `setPageCategory()` convention can replace this if DC wants stricter semantics later.

**2. `label` is now PII-scrubbed.** Was serializing the whole `eventData`; now `scrubPii()` recursively drops keys matching a deny-list (email, phone, name, address, DOB, SSN, etc.) before `JSON.stringify`. If every key is PII, the label slot is omitted entirely. Defensive floor — per-event allow-list can tighten this once DC confirms their preferred label schema.

Tests: 10/10 passing on the bridge (added 4 new cases: path-derived category, root fallback, recursive PII scrub, all-PII payload). Full suites green: 100/100 analytics, 639/639 portal web, typecheck clean.

Open question for DC (not blocking this PR): do they want page-view events to *also* fire as `['event', 'page_load', ...]` in addition to the `trackdynamic` call? Today only `trackdynamic` fires for page views — that's SiteImprove's recommended shape for SPA navigation, but the refined ticket reads as if both could be desired. Easy to add if confirmed.

**Comment by adbergen** [comment]
# QA Summary — DC-272

## What Changed

The DC portal now sends page-view and event data to SiteImprove. The integration mirrors what already exists for Mixpanel/Amplitude on DC: the SiteImprove script loads on DC pages only, and every SEBT page-load and tracked click ("CTA click") gets forwarded to SiteImprove as a `_sz.push()` call.

**No user-visible UI changes.** This is observability-only — the page looks and behaves identically. Verification happens via browser DevTools and the SiteImprove dashboard.

The mapping (per the refined ticket AC):
- **Action** = the SEBT event name (e.g. `page_load`, `cta_click`).
- **Category** = `flow:step` derived from the current page's analytics-route entry (e.g. `auth:verify_otp`, `address_update:address_form`, `cards:info`, `dashboard:dashboard`).
- **Label** = the event's payload as JSON, with known PII keys stripped recursively (email, phone, name, address, DOB, SSN, etc.).

## Affected Areas

- DC portal — every page (page-load tracking)
- DC portal — every tracked CTA button (sign-in, replacement card, address update, apply, etc.)
- CO portal — no behavior change (SiteImprove must NOT load for CO)

## Test Cases

### Happy Path

- [ ] **SiteImprove script loads on DC.**
  1. Sign in to the **DC portal** with DevTools open (Network tab, filter for "siteimprove").
  2. Navigate to any page.
  3. **Expected:** A network request to `siteimproveanalytics.com` (the SiteImprove JS snippet) loads successfully. No CSP violations or other errors appear in the browser console.

- [ ] **`_sz` queue contains a page-load command on each navigation.**
  1. Open DevTools Console.
  2. Navigate around the DC portal (login → verify → dashboard, etc.).
  3. After each page change, type `window._sz` into the console.
  4. **Expected:** Each navigation appends a `['trackdynamic', { url, ref, title }]` entry to the queue (or, after the SiteImprove script has booted, the call is delivered through the official `_sz.push` API and visible in the Network tab going to `siteimproveanalytics.io`).

- [ ] **CTA clicks forward as SiteImprove `event` commands with correct mapping.**
  1. In the DC portal, navigate to a page with a tracked CTA — e.g. the dashboard's "Request a replacement card" button.
  2. Watch the Network tab for outgoing SiteImprove requests, or watch `window._sz` push history.
  3. Click the CTA.
  4. **Expected:** A new entry of shape `['event', '<flow>:<step>', 'cta_click', '<json>']` is recorded.
     - Action is `cta_click`.
     - Category reflects the page you're on (e.g. `dashboard:dashboard`, `address_update:address_form`).
     - Label is a JSON string containing the CTA details but **not** any PII.

- [ ] **PII is scrubbed from the label.**
  1. On the DC portal, perform an action whose tracked event would normally carry PII (e.g. complete the ID-proofing form — the data layer event for that page does not carry PII directly, but for verification, inspect any tracked event after entering a name/email/SSN).
  2. Inspect the `_sz.push` history or the outgoing SiteImprove network call.
  3. **Expected:** No field in the JSON label contains email, phone, name (first/middle/last/full), street address, city, state, zip, DOB, or SSN. Other fields (e.g. `cta_id`, `cta_target`, `household_type`) are preserved.

### Edge Cases

- [ ] **An unmapped route still produces a usable category.**
  1. If you navigate to a page that isn't in the route map (or that doesn't set `page.flow`/`page.step`), trigger a CTA there.
  2. **Expected:** Category falls back to `unknown` — not blank, not a malformed string like `:`.

- [ ] **No errors when SiteImprove is blocked by an extension or network policy.**
  1. With an ad blocker or similar that blocks `siteimproveanalytics.com`, navigate the DC portal.
  2. **Expected:** The portal still works normally. `_sz` calls are queued in `window._sz` (the array form) but never delivered; no script errors propagate to the console.

### Regression Checks

- [ ] **SiteImprove does NOT load on the CO portal.**
  1. Sign in to the **CO portal** with DevTools open (Network tab).
  2. Navigate around several pages.
  3. **Expected:** Zero requests to `siteimproveanalytics.com` or `siteimproveanalytics.io`. `window._sz` is `undefined`. No script tags pointing at SiteImprove appear in the page source.

- [ ] **Mixpanel and Amplitude still receive events on DC.**
  1. With DevTools Network tab open, navigate the DC portal and click a tracked CTA.
  2. **Expected:** Mixpanel and Amplitude requests still fire alongside SiteImprove. None of the three blocks the others.

- [ ] **No new CSP violations on DC.**
  1. With DevTools Console open, navigate the DC portal across login → dashboard → cards → address.
  2. **Expected:** No "Refused to load script…" or "Refused to connect to…" CSP errors. The SiteImprove domains are in the allow-list when `NEXT_PUBLIC_SITEIMPROVE_ID` is set.

## Test Credentials

Use any seeded DC portal persona. Personas follow `sebt.dc+{scenario}@codeforamerica.org`.

- DC verified household — `sebt.dc+verified@codeforamerica.org` — OTP retrieval depends on environment (Mailpit locally; mailbox in deployed envs)
- DC co-loaded household — `sebt.dc+co-loaded@codeforamerica.org`
- DC household with no applications — `sebt.dc+not-started@codeforamerica.org`

For CO regression: any seeded CO persona, e.g. `sebt.co+co-loaded@codeforamerica.org`.

## Environment Notes

- SiteImprove is gated on **two** conditions: the portal must be running as `STATE=dc` AND the build-time env var `NEXT_PUBLIC_SITEIMPROVE_ID` must be set. If either is missing the SiteImprove integration is off.
- The deployed DC environments need `NEXT_PUBLIC_SITEIMPROVE_ID` configured (via the GitHub repo Variable and the Tofu config). Without it, the script won't load — please confirm with the platform/devops team that the value is set before testing in staging/prod.
- CO does not need the variable set; if it accidentally is, the state check still blocks SiteImprove from loading.
- The SiteImprove script and its event endpoint (`siteimproveanalytics.io`) have been added to the Content Security Policy as `script-src` and `connect-src` allow-list entries, but only when the env var is set.
- No new app-settings, AWS Secrets, or backend env vars — this is a frontend-only change.

File: packages/analytics/src/siteimprove-bridge.ts [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+  const flow = dl.get('page.flow', 'analytics', '') as string
+  const step = dl.get('page.step', 'analytics', '') as string
+  if (flow && step) return `${flow}:${step}`
+  return flow || step || 'unknown'
+}
+
+// Keys that may carry PII are dropped before the label is serialized. The
+// match is case-insensitive against the lowercased key. This is a defensive
+// floor — callers should still scope payloads to non-PII fields up front.
+const PII_KEYS = new Set([
```
Comment (necampanini) [inline packages/analytics/src/siteimprove-bridge.ts:39]: good list, and I think we can add a few other PII keys if we want to be even more exhaustive

- itin, 
- snapid / tanfid,
- medicaidid
- caseid 
- householdid

File: packages/analytics/src/siteimprove-bridge.ts [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+  'ssn',
+  'socialsecuritynumber'
+])
+
+function scrubPii(value: unknown): unknown {
+  if (value === null || typeof value !== 'object') return value
+  if (Array.isArray(value)) return value.map(scrubPii)
+  const out: Record<string, unknown> = {}
+  for (const [k, v] of Object.entries(value as Record<string, unknown>)) {
+    if (PII_KEYS.has(k.toLowerCase())) continue
```
Comment (necampanini) [inline packages/analytics/src/siteimprove-bridge.ts]: i think payloads using snake_case would get through this as written

`email_address, first_name, date_of_birth, social_security_number, phone_number, street_address, case_id`

File: src/SEBT.Portal.Web/src/app/layout.tsx [hunk 3]
```diff
@@ -157,6 +159,14 @@ export default async function RootLayout({
       )}
       {/* Amplitude - only rendered when NEXT_PUBLIC_AMPLITUDE_API_KEY is configured */}
       {amplitudeApiKey && <AmplitudeAnalytics apiKey={amplitudeApiKey} />}
+      {/* SiteImprove — DC-only per DC-272. Gated by both state and env var so an
+          accidentally-set NEXT_PUBLIC_SITEIMPROVE_ID in another state does not enable it. */}
+      {state === 'dc' && siteImproveId && (
```
Comment (necampanini) [inline src/SEBT.Portal.Web/src/app/layout.tsx:164]: worth adding a test for `next_public_state='co'` and making sure this section doesn't get rendered

File: packages/analytics/src/siteimprove-bridge.ts [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+
+    // Label carries the JSON-serialized event payload with known PII keys
+    // dropped (see PII_KEYS). When DC confirms a per-event label scheme this
+    // can be tightened to a per-event allow-list.
+    const category = deriveCategory(dl)
+    const scrubbed = detail.eventData ? (scrubPii(detail.eventData) as Record<string, unknown>) : undefined
+    const hasData = scrubbed && Object.keys(scrubbed).length > 0
+    pushToSz(
+      hasData
+        ? ['event', category, detail.eventName, JSON.stringify(scrubbed)]
```
Comment (necampanini) [inline packages/analytics/src/siteimprove-bridge.ts]: probably safe as is, but if circular refs or BigInts appear in `scrubbed` (not sure why they would), this would throw

**Review by necampanini — CHANGES_REQUESTED** [review]
looks great, just some "belt and suspenders" changes for PII requested

**Review by necampanini — APPROVED** [review]
